src/Lean/Data/Name.lean

```diff
@@ -65,26 +65,50 @@ def isSuffixOf : Name → Name → Bool
   | num p₁ n₁ _, num p₂ n₂ _ => n₁ == n₂ && isSuffixOf p₁ p₂
   | _,           _           => false
 
-def lt : Name → Name → Bool
-  | anonymous,   anonymous   => false
-  | anonymous,   _           => true
-  | num p₁ i₁ _, num p₂ i₂ _ => lt p₁ p₂ || (p₁ == p₂ && i₁ < i₂)
-  | num _ _ _,   str _ _ _   => true
-  | str p₁ n₁ _, str p₂ n₂ _ => lt p₁ p₂ || (p₁ == p₂ && n₁ < n₂)
-  | _,           _           => false
-
-def quickLtAux : Name → Name → Bool
-  | anonymous, anonymous   => false
-  | anonymous, _           => true
-  | num n v _, num n' v' _ => v < v' || (v = v' && n.quickLtAux n')
-  | num _ _ _, str _ _ _   => true
-  | str n s _, str n' s' _ => s < s' || (s = s' && n.quickLtAux n')
-  | _,         _           => false
+def cmp : Name → Name → Ordering
+  | anonymous,   anonymous   => Ordering.eq
+  | anonymous,   _           => Ordering.lt
+  | _,           anonymous   => Ordering.gt
+  | num p₁ i₁ _, num p₂ i₂ _ => 
+    match cmp p₁ p₂ with  
+    | Ordering.eq => compare i₁ i₂
+    | ord => ord
+  | num _ _ _,   str _ _ _   => Ordering.lt
+  | str _ _ _,   num _ _ _   => Ordering.gt
+  | str p₁ n₁ _, str p₂ n₂ _ => 
+    match cmp p₁ p₂ with 
+    | Ordering.eq => compare n₁ n₂
+    | ord => ord
+    
+def lt (x y : Name) : Bool :=
+  match cmp x y with
+  | Ordering.lt => true
+  | _ => false
```

Comment:
```suggestion
  cmp x y == Ordering.lt
```
should work too, right?

Reply:
I guess it doesn't, but then we should make it work

Reply:
sure it works, just needed the `BEq` deriving. I call it a case of "Wald vor lauter Bäumen nicht gesehen".